src/Utilities/FlowAnalysis/FlowAnalysis/Analysis/TaintedDataAnalysis/ZipSlipSanitizers.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Analyzer.Utilities.PooledObjects;
+
+namespace Analyzer.Utilities.FlowAnalysis.Analysis.TaintedDataAnalysis
+{
+    internal static class ZipSlipSanitizers
+    {
+        /// <summary>
+        /// <see cref="SanitizerInfo"/>s for zip slip tainted data sanitizers.
+        /// </summary>
+        public static ImmutableHashSet<SanitizerInfo> SanitizerInfos { get; }
+
+        static ZipSlipSanitizers()
+        {
+            var builder = PooledHashSet<SanitizerInfo>.GetInstance();
+
+            builder.AddSanitizerInfo(
+                WellKnownTypeNames.SystemIOPath,
+                isInterface: false,
+                isConstructorSanitizing: false,
+                sanitizingMethods: new[] {
+                    "GetFileName",
+                });
+            builder.AddSanitizerInfo(
+                WellKnownTypeNames.SystemString,
+                isInterface: false,
+                isConstructorSanitizing: false,
+                sanitizingMethods: new[] {
+                    "StartsWith",
+                    "Substring",
+                });
+
```

Comment:
If I remember correctly, TaintedDataAnalysis's current implementation is just going to sanitize the return value (the boolean return values in this case), not the potentially tainted data passed in.

Reply:
You're welcome to extend the functionality if you want.  :-D

---
In reply to: [290959865](https://github.com/dotnet/roslyn-analyzers/pull/2521#discussion_r290959865) [](ancestors = 290959865)